File: CashuWallet/Models/Tokens/TokenInfo.swift
import Foundation

struct TokenInfo {
    let amount: UInt64
    let mint: String
    let unit: String
    let memo: String?
    let proofCount: Int
    
    /// Parse a cashu token string
    static func parse(_ tokenString: String) -> TokenInfo? {
        TokenParser.tokenInfo(from: tokenString)
    }
}

File: android/app/src/main/java/org/cashu/wallet/Models/Tokens/TokenInfo.kt
package org.cashu.wallet.Models

import kotlinx.serialization.Serializable
import org.cashu.wallet.Core.TokenParser

@Serializable
data class TokenInfo(
    val amount: Long,
    val mint: String,
    val unit: String,
    val memo: String?,
    val proofCount: Int,
) {
    companion object {
        fun parse(tokenString: String): TokenInfo? = TokenParser.tokenInfo(tokenString)
    }
}
